### DISCUSSION TITLE
`fetcher.submit` sends request twice in `useEffect`

### DISCUSSION THREAD

**Discussion by kicut** [remix-run/remix#4469] [Q&A] [answered]
Hello everyone, 

How do I prevent the `fetcher.submit` in my `Score` component from sending the request twice? Is it possible to prevent it like using AbortController?

```jsx
function Score({ payload }) {
  const fetcher = useFetcher();

  useEffect(() => {
    // 🚨 sends request twice
    fetcher.submit(payload, { method: "post" });
  }, []);

  // ...
}

function Game() {
  const { ended } = useCountdown();
  const [score, setScore] = useState({});

  return !ended ? (
    <GamePlay setScore={setScore} />
  ) : (
    <Score score={score} />
  );
}

```

**Comment by machour** [comment, marked as answer]
if you're using React 18, then this is an expected behavior.
See workaround proposed by Dan here: https://github.com/facebook/react/issues/24502#issuecomment-1118867879
To me, it feels better than starting a fetch and aborting it
